proxy/proxy.go

```diff
@@ -705,8 +705,8 @@ func WithParams(p Params) *Proxy {
 	if p.EndpointRegistry == nil {
 		p.EndpointRegistry = routing.NewEndpointRegistry(routing.RegistryOptions{})
 	}
-
 	hostname := os.Getenv("HOSTNAME")
+	p.Routing.Metrics = m
```

Comment:
This does not look right - it modifies proxy parameters.

It should be somewhere in skipper.go like
```go
routing := routing.New(ro)
defer routing.Close()

routing.Metrics = mtr
```

Reply:
great point, this makes the implementation even simplier!